cmake/Macros.cmake

```diff
@@ -25,17 +25,20 @@ macro(now RESULT)
   endif(WIN32)
 endmacro(now)
 
+find_package(Git)
+
 macro(git_id RESULT)
-  if(WIN32)
-    execute_process(COMMAND "cmd" " /C git --git-dir=${PROJECT_SOURCE_DIR}/.git rev-parse --short=8 HEAD" OUTPUT_VARIABLE ${RESULT})
-    string(REGEX REPLACE "(.*)\n" "\\1" ${RESULT} ${${RESULT}})
-  elseif(UNIX)
-    execute_process(COMMAND "git" "--git-dir=${PROJECT_SOURCE_DIR}/.git" "rev-parse" "--short=8" "HEAD" OUTPUT_VARIABLE ${RESULT})
+  if (GIT_FOUND)
+    if(WIN32)
+      execute_process(COMMAND "cmd" " /C ${GIT_EXECUTABLE} --git-dir=${PROJECT_SOURCE_DIR}/.git rev-parse --short=8 HEAD" OUTPUT_VARIABLE ${RESULT})
+    else()
+      execute_process(COMMAND "${GIT_EXECUTABLE}" "--git-dir=${PROJECT_SOURCE_DIR}/.git" "rev-parse" "--short=8" "HEAD" OUTPUT_VARIABLE ${RESULT})
+    endif()
     string(REGEX REPLACE "(.*)\n" "\\1" ${RESULT} ${${RESULT}})
-  else(WIN32)
-    message(SEND_ERROR "Git ID implemented")
+  else()
+    message(WARNING "Git executable not found, revision number not available.")
     set(${RESULT} 0)
-  endif(WIN32)
+  endif()
```

Comment:
I like this change, I did a more cumbersome solution in https://github.com/opentx/opentx/commit/84ebeb01c34122c3959f2b55d3592c440d93a96d.

Could you add one more enhancement: If we build from a source that is not git repository (ie from downloaded zip archive), detect that and also use 0 for git_id. But first we have to test how this behaves in such case (no git files).